presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -247,6 +251,12 @@ public String getQueryHash()
         return queryHash;
     }
 
+    @JsonProperty
+    public Optional<String> getQueryTemplateHash()
+    {
+        return queryTemplate.isPresent() ? Optional.of(computeQueryHash(queryTemplate.get())) : Optional.empty();
```

Comment:
It would be better style to use Optional.map here

Reply:
Updated to use Optional.map :)